src/uu/mv/src/mv.rs

```diff
@@ -732,14 +749,21 @@ fn rename(
                 return Ok(());
             }
             OverwriteMode::Interactive => {
-                if !prompt_yes!(
-                    "{}",
-                    translate!("mv-prompt-overwrite", "target" => to.quote())
-                ) {
+                let prompt_msg = get_interactive_prompt(to);
+                if !prompt_yes!("{}", prompt_msg) {
                     return Err(io::Error::other(""));
                 }
             }
             OverwriteMode::Force => {}
+            OverwriteMode::Default => {
+                // GNU mv prompts when stdin is a TTY and target is not writable
+                if std::io::stdin().is_terminal() && !is_writable(to) {
+                    let prompt_msg = get_interactive_prompt(to);
+                    if !prompt_yes!("{}", prompt_msg) {
+                        return Err(io::Error::other(""));
+                    }
```

Comment:
since this is duplicated from above, what about extracting it into a function or simple closure ?